components/x-teaser-timeline/src/lib/transform.js

```diff
@@ -165,15 +165,21 @@ export const buildModel = ({
 		latestItemsTime,
 		latestItemsAgeHours
 	})
-
 	if (itemGroups.length > 0 && customSlotContent) {
-		const insertPosition = Math.min(customSlotPosition, items.length)
-		const insert = getGroupAndIndex(itemGroups, insertPosition)
-		const copyOfItems = [...itemGroups[insert.group].items]
+		const customSlotContentArray = Array.isArray(customSlotContent) ? customSlotContent : [customSlotContent]
+		const customSlotPositionArray = Array.isArray(customSlotPosition)
+			? customSlotPosition
+			: [customSlotPosition]
+
+		for (const [index, slotContent] of customSlotContentArray.entries()) {
+			const insertPosition = Math.min(customSlotPositionArray[index], items.length + index)
+			const insert = getGroupAndIndex(itemGroups, insertPosition)
+			const copyOfItems = [...itemGroups[insert.group].items]
 
-		copyOfItems.splice(insert.index, 0, customSlotContent)
+			copyOfItems.splice(insert.index, 0, slotContent)
 
-		itemGroups[insert.group].items = copyOfItems
+			itemGroups[insert.group].items = copyOfItems
+		}
```

Comment:
Suggestion [non-blocking]: extracting this into a named function might make the purpose of this loop easier to grok at a glance

Reply:
I think I've done this.